### PR TITLE
Add examples to scipy.interpolate.spalde

### PR DESCRIPTION
**Pull request by mabatalla**
#### Reference issue
Part of DOC: Add "Examples" to docstrings https://github.com/scipy/scipy/issues/7168

#### What does this implement/fix?
Adds example for interpolate.spalde

[skip actions] [skip cirrus]

### CODE REVIEW

File: scipy/interpolate/_fitpack_py.py [hunk 1]
```diff
@@ -564,6 +564,30 @@ def spalde(x, tck):
     .. [3] Dierckx P. : Curve and surface fitting with splines, Monographs on
        Numerical Analysis, Oxford University Press, 1993.
 
+    Examples
+    --------
+    >>> from scipy.interpolate import spalde
+    >>> x = (0.5)
```
Comment (ev-br) [inline scipy/interpolate/_fitpack_py.py:570]:
```suggestion
    >>> x = 0.5
```

**Review by j-bowhay — CHANGES_REQUESTED** [review]
Could you add a figure? I think this example would be nice if it had a visual aspect

**Comment by ev-br** [comment]
Would be nice to demonstrate that `spalde` is equivalent to calling `splev` with varying derivative orders

File: scipy/interpolate/_fitpack_py.py (hunk 1 above)
Comment (mabatalla) [reply scipy/interpolate/_fitpack_py.py:570]: I will modify the file with this and figure sugestion

**Comment by astrojuanlu** [comment]
Nice figure 😄 

image

https://output.circle-artifacts.com/output/job/48e61197-6507-4f98-bfc3-56b7609fa84d/artifacts/0/html/reference/generated/scipy.interpolate.spalde.html#scipy.interpolate.spalde

File: scipy/interpolate/_fitpack_py.py [hunk 2]
```diff
@@ -564,6 +567,97 @@ def spalde(x, tck):
     .. [3] Dierckx P. : Curve and surface fitting with splines, Monographs on
        Numerical Analysis, Oxford University Press, 1993.
 
+    Examples
+    --------
+    >>> import numpy as np
+    >>> import matplotlib.pyplot as plt
+    >>> from scipy.interpolate import BSpline, spalde, splder, splev
+    
+    >>> # Store characteristic parameters of a Bspline =================================
```
[the remark this answers is not in the document]
Comment (j-bowhay) [reply scipy/interpolate/_fitpack_py.py:576]: Here and other places

File: scipy/interpolate/_fitpack_py.py [hunk 3]
```diff
@@ -564,6 +567,97 @@ def spalde(x, tck):
...
+    >>> ax3.plot(x, m3_y_deriv1, '--', color='#999999', label='1st deriv.')
+    >>> ax3.plot(x, m3_y_deriv2, '-.', color='#999999', label='2nd deriv.')
+    >>> ax3.plot(x, m3_y_deriv3, ':', color='#999999', label='3rd deriv.')
+    >>> # Plot a reference line
+    >>> ax3.vlines(x_ref, -11, 10.0, color='r', label=f'x = {x_ref}')
+    >>> # Aditional info
+    >>> ax3.set_title('spalde()')
+    >>> ax3.legend()
+    >>> # Plot -------------------------------------------------------------------------
+    >>> plt.show()
```
Comment (j-bowhay) [inline scipy/interpolate/_fitpack_py.py:641]: I think this might fix the sizing
```suggestion
    >>> plt.tight_layout()
    >>> plt.show()
```

File: scipy/interpolate/_fitpack_py.py [hunk 4]
```diff
@@ -564,6 +567,97 @@ def spalde(x, tck):
...
+    ...        (0, 0, 0, 6, 0, 0, 0),
+    ...        3)
+    
+    >>> # Instance a B-spline object ===================================================
+    >>> # `BSpline` objects are prefered, except for spalde()
+    >>> bspl = BSpline(tck[0], tck[1], tck[2])
+    >>> # Generate extra points to get a smooth curve ----------------------------------
+    >>> x = np.linspace(min(tck[0]), max(tck[0]), 100)
+    >>> x_ref = 1.25
+
```
Comment (j-bowhay) [inline scipy/interpolate/_fitpack_py.py:588]: Could you add some written text explaining why we are showing these three methods

File: scipy/interpolate/_fitpack_py.py [hunk 5]
```diff
@@ -354,6 +354,9 @@ def splev(x, tck, der=0, ext=0):
     --------
     Examples are given :ref:`in the tutorial <tutorial-interpolate_splXXX>`.
 
+    A comparison between splev, splder and spalde to compute the derivatives of a 
+    B-spline can be found in the spalde entry.
```
Comment (j-bowhay) [inline scipy/interpolate/_fitpack_py.py:358]:
```suggestion
    A comparison between `splev`, `splder` and `spalde` to compute the derivatives of a 
    B-spline can be found in the `spalde` examples section.
```

**Review by j-bowhay — CHANGES_REQUESTED** [review]
Thanks @mabatalla I think this is looking like a nice example. A few suggestions to bring this closer in line with the style used throughout scipy

File: scipy/interpolate/_fitpack_py.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mabatalla) [reply scipy/interpolate/_fitpack_py.py:576]: Should I delete all the ------ or just the ======

File: scipy/interpolate/_fitpack_py.py [hunk 6]
```diff
@@ -564,6 +567,97 @@ def spalde(x, tck):
...
+    >>> tck = ((-2, -2, -2, -2, -1, 0, 1, 2, 2, 2, 2),
+    ...        (0, 0, 0, 6, 0, 0, 0),
+    ...        3)
+    
+    >>> # Instance a B-spline object ===================================================
+    >>> # `BSpline` objects are prefered, except for spalde()
+    >>> bspl = BSpline(tck[0], tck[1], tck[2])
+    >>> # Generate extra points to get a smooth curve ----------------------------------
+    >>> x = np.linspace(min(tck[0]), max(tck[0]), 100)
+    >>> x_ref = 1.25
```
[the remark this answers is not in the document]
Comment (mabatalla) [reply scipy/interpolate/_fitpack_py.py:587]: By reading the docs I got that Bspline objects are preferred rather than tck-tuples except for spalde() and I thought it enriches the sample

[the remark this answers is not in the document]
Comment (j-bowhay) [reply scipy/interpolate/_fitpack_py.py:587]: Sorry I meant what is the significance of `x_ref`

**Comment by ev-br** [comment]
refguide-check failures are real: https://app.circleci.com/pipelines/github/scipy/scipy/23229/workflows/ff5a9a92-72dd-4ac2-bf0d-d03533a78c1f/jobs/78150

Basically, it's a bad idea to `print` stuff in docstring examples. This is of course fixable, but --- just take a look at the rendered docs:
https://output.circle-artifacts.com/output/job/2dfad60b-f9af-48e0-8783-27188b0c5fad/artifacts/0/html/reference/generated/scipy.interpolate.spalde.html#scipy.interpolate.spalde

We would then have a wall of code (mostly trivial, why do we need to specify the colors? Is MPL default palette not ok? Why do we specify the colors on each line? Labels are great, but can we not loop over a `zip(orders, labels)` or even `enumerate(labels)`). Then we have a row of small plots, and then a wall of code again.

TL;DR: would be great to make docstring additions leaner.

**Comment by mabatalla** [comment]
Not TL at all @ev-br, I appreciate your time and advice. This might be a silly sample, but I am commited to return to the open-source communty as much as I can. Hope this is the first but not least PR.

Saying that, I have simplified the code avoiding the trivial parts. I always prefer linetypes rather than colors to avoid colorblind/color reproduction problems, but using the gray was just aesthetic.

Hope next try is ok!

File: scipy/interpolate/_fitpack_py.py [hunk 7]
```diff
@@ -564,6 +567,66 @@ def spalde(x, tck):
...
+    In this example, we will demonstrate that `spalde` is equivalent to
+    calling `splev` and `splder`.
+    
+    >>> import numpy as np
+    >>> import matplotlib.pyplot as plt
+    >>> from scipy.interpolate import BSpline, spalde, splder, splev
+    
+    >>> # Store characteristic parameters of a B-spline
+    >>> # (knots, coefficients, degree of the spline)
+    >>> tck = ((-2, -2, -2, -2, -1, 0, 1, 2, 2, 2, 2),
```
Comment (ev-br) [inline scipy/interpolate/_fitpack_py.py:582]:
```suggestion
    >>> tck = ((-2, -2, -2, -2, -1, 0, 1, 2, 2, 2, 2),      # knots
```

File: scipy/interpolate/_fitpack_py.py [hunk 8]
```diff
@@ -564,6 +567,66 @@ def spalde(x, tck):
...
+    ...        (0, 0, 0, 6, 0, 0, 0),
```
Comment (ev-br) [inline scipy/interpolate/_fitpack_py.py:583]:
```suggestion
    ...        (0, 0, 0, 6, 0, 0, 0),   # coefficients
```

File: scipy/interpolate/_fitpack_py.py [hunk 9]
```diff
@@ -564,6 +567,66 @@ def spalde(x, tck):
...
+    ...        3)
```
Comment (ev-br) [inline scipy/interpolate/_fitpack_py.py:584]:
```suggestion
    ...        3)
```
```suggestion
    ...        3)        # cubic
```

File: scipy/interpolate/_fitpack_py.py [hunk 10]
```diff
@@ -564,6 +567,66 @@ def spalde(x, tck):
...
+    >>> m2_y_deriv3 = splder(bspl, n=3)(x)
+    >>> # Method 3: spalde()
+    >>> m3_y_bsplin = [spalde(i, tck)[0] for i in x ]
+    >>> m3_y_deriv1 = [spalde(i, tck)[1] for i in x ]
+    >>> m3_y_deriv2 = [spalde(i, tck)[2] for i in x ]
+    >>> m3_y_deriv3 = [spalde(i, tck)[3] for i in x ]
+    
+    Graphic comparison
+    
+    >>> # Generate a figure with three axes for graphic comparison
```
Comment (ev-br) [inline scipy/interpolate/_fitpack_py.py:612]: This sentence is not needed really, given that you actually generate said figure just below.

File: scipy/interpolate/_fitpack_py.py [hunk 11]
```diff
@@ -564,6 +567,66 @@ def spalde(x, tck):
...
+    >>> # Method 2: splder()
+    >>> m2_y_bsplin = splder(bspl, n=0)(x)
+    >>> m2_y_deriv1 = splder(bspl, n=1)(x)
+    >>> m2_y_deriv2 = splder(bspl, n=2)(x)
+    >>> m2_y_deriv3 = splder(bspl, n=3)(x)
+    >>> # Method 3: spalde()
+    >>> m3_y_bsplin = [spalde(i, tck)[0] for i in x ]
+    >>> m3_y_deriv1 = [spalde(i, tck)[1] for i in x ]
+    >>> m3_y_deriv2 = [spalde(i, tck)[2] for i in x ]
+    >>> m3_y_deriv3 = [spalde(i, tck)[3] for i in x ]
```
Comment (ev-br) [inline scipy/interpolate/_fitpack_py.py:608]: This part should be shortened. Maybe include just one one derivative order to show the three ways.

**Review by ev-br — CHANGES_REQUESTED** [review]
This still looks like a wall of code, I'm afraid. Also, doc build and refgude-check failures need looking at for this doc-only change.

**Comment by j-bowhay** [comment]
@mabatalla friendly ping, there are some comments here for you, are you interested in finishing this off?

**Comment by mabatalla** [comment]
> @mabatalla friendly ping, there are some comments here for you, are you interested in finishing this off?

Sure!! I expect to go back to the PR in a few days!

**Comment by mabatalla** [comment]
Hello there! @j-bowhay @ev-br, had trouble building scipy-dev and tried with a fresh start. I've modified the file and passed the tests again. Can you please give me further information on this "doc build and refgude-check failures", I won't send a ned PR without testing everything. P.S. @astrojuanlu maybe you can also help. Thx in advance to everyone!

**Comment by ev-br** [comment]
Just run `$ python dev.py refguide-check -s interpolate` locally.
